src/psi.rs

```diff
@@ -275,3 +281,45 @@ fn hash_2(input1: u64, input2: Fq) -> Fq {
 
     hash
 }
+
+/// given x coords construct Li polynomials
+fn construct_lagrange_basis<F: FftField>(evaluation_domain: &[F]) -> Vec<DensePolynomial<F>> {
+    let mut bases = Vec::with_capacity(evaluation_domain.len());
+    for i in 0..evaluation_domain.len() {
+        let mut l_i = DensePolynomial::from_coefficients_slice(&[F::one()]);
+        let x_i = evaluation_domain[i];
+        for (j, _) in evaluation_domain.iter().enumerate() {
+            if j != i {
+                let xi_minus_xj_inv = (x_i - evaluation_domain[j]).inverse().unwrap();
+                l_i = &l_i
+                    * &DensePolynomial::from_coefficients_slice(&[
+                        -evaluation_domain[j] * xi_minus_xj_inv,
+                        xi_minus_xj_inv,
+                    ]);
+            }
+        }
+
+        bases.push(l_i);
+    }
+
+    bases
+}
+
+
+#[test]
+fn test_interpolation_lib() {
+    let n: usize = 32;
+    let mut rng = test_rng();
+
+    let roots: Vec<_> = (0..n).map(|_| Fr::rand(&mut rng)).collect();
+    
+    let lagrange_basis = construct_lagrange_basis(&roots);
+    let f_evals: Vec<_> = (0..n).map(|_| Fr::rand(&mut rng)).collect();
+
+    let mut f_slow = DensePolynomial::default();
+    for (li, &fi) in lagrange_basis.iter().zip(f_evals.iter()) {
+        f_slow += (fi, li);
+    }
+
```

Comment:
should have some assertion here? i see u were probably just checking by eye but might as well add assertions so if you break this, the test will fail